buku.py

```diff
@@ -2182,6 +2184,107 @@ def sigint_handler(signum, frame):
 signal.signal(signal.SIGINT, sigint_handler)
 
 
+# ---------------------
+# Editor mode functions
+# ---------------------
+
+def open_editor(editors, url, title_in, tags_in, desc):
+    editor = None
+    if editors is not None and len(editors) > 0:
+        editor = editors[0]
+    else:
+        editor = os.environ.get('EDITOR', None)
+
+    if editor is None:
+        logerr("editor args not provided and $EDITOR env var is not set.")
+        logerr("Operation aborted.")
+        return None
+
+    temp_file_content = to_temp_file_content(url, title_in, tags_in, desc)
+
+    with tempfile.NamedTemporaryFile(mode="w+", suffix='.tmp', encoding="utf-8") as temp:
+        temp.write(temp_file_content)
+        temp.flush()
+        subprocess.call([editor, temp.name])
+        with open(temp.name) as f: # for some reason sometimes this don't get saved properly
```

Comment:
Sorry, i think this wasn't clear.

The problem is that after you save your editors, sometimes the temp object doesn't get updated with the new content. The workaround I did here is to open the temp file again and read it, instead of using the temp object.